### PR TITLE
Case 21037: Allow the script engine to use FileSeclector when loading scriopts

### PR DESCRIPTION
**Pull request by danteruiz**
Allow for the ability for the script engine to use the fileselector to load platform specific scripts 

ticket - https://highfidelity.manuscript.com/f/cases/21037/Quest-allow-for-platform-specific-default-scripts

### CODE REVIEW

File: libraries/shared/src/shared/FileUtils.cpp [hunk 1]
```diff
@@ -30,6 +30,11 @@ const QStringList& FileUtils::getFileSelectors() {
     static std::once_flag once;
     static QStringList extraSelectors;
     std::call_once(once, [] {
+
+#if defined(Q_OS_ANDROID)
+        //extraSelectors << "android_" HIFI_ANDROID_APP;
```
Comment (danteruiz) [inline libraries/shared/src/shared/FileUtils.cpp]: this will be used once this code is merged into the feature/quest branch

**Comment by danteruiz** [comment]
Test plan
- make sure that the default app appear in the tablet in desktop and hmd mode

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17049/HighFidelity-Beta-PR14854-30bd977.apk

File: libraries/script-engine/src/ScriptEngines.cpp [hunk 2]
```diff
@@ -476,6 +477,8 @@ ScriptEnginePointer ScriptEngines::loadScript(const QUrl& scriptFilename, bool i
         scriptUrl = normalizeScriptURL(scriptFilename);
     }
 
+    scriptUrl = QUrl(FileUtils::selectFile(scriptUrl.toString()));
```
Comment (jherico) [inline libraries/script-engine/src/ScriptEngines.cpp]: I don't understand how this works if the script url is a network URL

File: libraries/shared/src/shared/FileUtils.cpp (hunk 1 above)
Comment (jherico) [reply libraries/shared/src/shared/FileUtils.cpp]: we also want to use it now, to migrate all the existing +android stuff to +android_interface

**Comment by jherico** [comment]
The test plan also need to cover loading scripts from network URLs and QRC urls as well as the local filesystem.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17049/HighFidelity-Beta-PR14854-30bd977.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17049/HighFidelity-Beta-Interface-PR14854-30bd977.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17049/HighFidelity-Beta-PR14854-30bd977.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17049/HighFidelity-Beta-Interface-PR14854-30bd977.exe

**Comment by JeffClinton** [comment]
Currently testing

**Pull request merged**